schemas/iso19115-3.2018/src/main/plugin/iso19115-3.2018/convert/fromJsonOpenDataSoft.xsl

```diff
@@ -264,15 +268,84 @@
               </mri:pointOfContact>
             </xsl:for-each>
 
-            <!-- ODS themes copied as topicCategory -->
-            <xsl:if test="metas/theme">
-                <xsl:for-each select="metas/theme">
-                  <mri:topicCategory>
-                    <mri:MD_TopicCategoryCode>
-                      <xsl:value-of select="."/>
-                    </mri:MD_TopicCategoryCode>
-                  </mri:topicCategory>
-                </xsl:for-each>
+            <xsl:variable name="odsThemeToIsoTopic" as="node()*">
```

Comment:
This mapping looks super verbose; maybe something like https://github.com/geonetwork/core-geonetwork/blob/fa9d324aff38e0faf516c317e6cff4060633eece/schemas/iso19115-3.2018/src/main/plugin/iso19115-3.2018/convert/protocol-mapping.xsl#L12 would be more compact and readable?

Reply:
Especially since several themes are mapped to the same topic

Reply:
Also this will only work for FR and EN, I don't remember if that was a problem or not?

Reply:
Lastly, I would put this mapping in another file, to not make this XSL file too big